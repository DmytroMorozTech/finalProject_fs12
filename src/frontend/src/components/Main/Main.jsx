import React from "react";
import {Profile} from "./Profile/Profile";
import {Feed} from "./Feed/Feed";
import {Favorites} from "./Favorites/Favorites";