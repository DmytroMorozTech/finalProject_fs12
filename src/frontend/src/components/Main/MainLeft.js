import React from 'react'
import {StyleMainLeft} from './StyleMainLeft'

function MainLeft () {
  const classes = StyleMainLeft()

  return (
    <div classes={classes.container}>
      <div className={classes.profile_card}>
        <div className={classes.profile_info}>
          <div className={classes.profile_bg}>
            <a>
              <div className={classes.profile_pic}/>
              <link className>Ivan Ivanov</link>
            </a>
            <a>
              <div className={classes.profile_info__text}>Junior C++ Programmer</div>
            </a>
          </div>
        </div>
        <div className={classes.profile_widget}>
          <a>
            <div className={classes.profile_widget__span}>
              <span>Contacts</span>
              <span>Add your contacts here</span>
            </div>
            <img src="" alt=""/>
          </a>
        </div>
        <div className={classes.profile_item}>
          <span>
            <img src="" alt=""/>
                        My Code
          </span>
        </div>
      </div>
      <div className={classes.social_card}>
        <a>
          <span>Groups and communities</span>
        </a>
        <a>
          <span>Events and meetings</span>
          <img src="" alt=""/>
        </a>
        <a>
          <span>Followed hashtags</span>
        </a>
        <a>
          <span>Discover more</span>
        </a>
      </div>
    </div>
  )
}

export default MainLeft