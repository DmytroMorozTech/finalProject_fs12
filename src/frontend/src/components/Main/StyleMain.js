import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({
  mainContainer: {
    display: 'grid',
    gridArea: 'main'
  }
}))