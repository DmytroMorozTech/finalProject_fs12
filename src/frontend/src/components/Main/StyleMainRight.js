import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({
  container: {
    display: 'grid',
    gridArea: 'rightside'
  },
  followCard: {
    textAlign: 'center',
    overflow: 'hidden',
    backgroundColor: 'white',
    position: 'relative',
    border: 'none',
    borderRadius: '4.5px',
    padding: '11.5px',
    boxShadow: '0 0 0 1px rgb(0 0 0 / 15%), 0 0 0 rgb(0 0 0 / 20%)'
  },
  followCardTitle: {
    display: 'inline-flex',
    fontSize: '15.5px',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    color: 'rgba(0, 0, 0, 0.5)'
  },
  followCardNewsList: {
    marginTop: '16px'
  },
  newsListItems: {
    display: 'flex',
    alignItems: 'center',
    position: 'relative',
    margin: '12px 0'
  },
  newsListItem: {
    display: 'flex',
    flexDirection: 'column'
  },
  btnNewsListItem: {
    backgroundColor: 'transparent',
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    outline: 'none',
    textAlign: 'center',
    padding: '16px',
    borderRadius: '14px',
    maxWidth: '475px',
    maxHeight: '31.5px',
    boxSizing: 'border-box',
    fontWeight: '550'
  },
  text: {
    fontSize: '11px'
  },
  recommendationsText: {
    display: 'flex',
    color: '#0a66c2',
    fontSize: '14px',
    alignItems: 'center'
  }
}))