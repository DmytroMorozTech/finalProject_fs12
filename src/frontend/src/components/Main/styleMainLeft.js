import { makeStyles } from '@material-ui/core/styles'

export default makeStyles(() => ({
  container: {
    display: 'grid',
    gridArea: 'leftside'
  },

  profileCard: {
    textAlign: 'center',
    overflow: 'hidden',
    marginBottom: '7.5px',
    boxShadow: '0 0 0 1px rgb(0 0 0 / 15%), 0 0 0 rgb(0 0 0 / 19.5%)',
    borderRadius: '4.5px',
    border: 'none',
    position: 'relative',
    transition: 'box-shadow 85ms'
  },

  profileInfo: {
    wordBreak: 'break-word',
    borderBottom: '1px solid rgba(0, 0, 0, 0.5)',
    wordWrap: 'break-word',
    padding: '12px 12px 15.5px'
  },

  profileBg: {
    background: 'center, no-repeat, 462px',
    height: '53.5px',
    margin: '-11.5px -11.5px 0'
  },

  profilePic: {
    background: 'center, white, content-box, 60%, no-repeat',
    border: '1.5px solid #fff',
    boxSizing: 'border-box',
    boxShadow: 'none',
    width: '71.5px',
    height: '71.5px',
    borderRadius: '29.5%',
    margin: '-37.5px auto 11.5px'
  },

  profileInfoText: {
    marginTop: '3.5px',
    color: '#0a66c2',
    fontSize: '12px',
    fontWeight: '390',
    lineHeight: '1.32'
  },

  profileWidget: {
    paddingTop: '11.5px',
    paddingBottom: '11.5px',
    borderBottom: '1px solid rgba(0, 0, 0, 0.12)',
    'a': {
      display: 'flex',
      textDecoration: 'none',
      alignItems: 'center',
      justifyContent: 'space-between',
      padding: '3.5px 11.5px',
      '&:hover': {
        backgroundColor: 'rgba(0, 0, 0, 0.9)'
      }
    }
  },

  profileWidgetText: {
    display: 'flex',
    flexDirection: 'column',
    textAlign: 'left',
    fontSize: '12px',
    lineHeight: '1.33',
    '&:first-child': {
      color: 'rgba(0, 0, 0, 0.6)',
    },
    '&:nth-child(2)': {
      color: 'rgba(0, 0, 0, 0.11)'
    }
  },

  profileItem: {
    borderColor: 'rgba(0, 0, 0, 0.8)',
    textAlign: 'left',
    padding: '12px',
    fontSize: '12px',
    display: 'block',
  },

  socialCard: {
    display: 'flex',
    flexDirection: 'column',
    textAlign: 'left',
    padding: '7.5px 0 0',
    'a': {
      color: 'black',
      fontSize: '11.5px',
      padding: '3.5px 11.5px 3.5px 11.5px',
      '&:hover': {
        color: '#0a66c2',
      },
      'span': {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
      },
      '&:last-child': {
        textDecoration: 'none',
        color: 'rgba(0, 0, 0, 0.5)',
        padding: '11.5px',
        borderTop: '1px solid #d6cec2',
        '&:hover': {
          backgroundColor: 'rgba(0, 0, 0, 0.09)'
        }
      }
    } 
  },
  
  profileLink: {
    color: 'rgba(0, 0, 0, 0.5)',
    fontSize: '14px',
    lineHeight: '1.5',
    fontWeight: '560'
  }
}))
