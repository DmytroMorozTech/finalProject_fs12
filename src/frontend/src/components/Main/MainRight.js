import React from 'react'
import StyleMainRight from './StyleMainRight'

function MainRight () {
  const classes = StyleMainRight()

  return (
    <div className={classes.container}>
      <div className={classes.followCard}>
        <div className={classes.followCardTitle}>
          <h2>Add to your feed</h2>
        </div>
        <div className={classes.followCardNewsList}>
          <li className={classes.newsListItems}>
            <a>
              <div className={classes.newsPic}/>
            </a>
            <div className={classes.newsListItem}>
              <span className={classes.text}>#LinkedIn</span>
              <button className={classes.btnNewsListItem}>Follow</button>
            </div>
          </li>
          <li className={classes.newsListItems}>
            <a>
              <div className={classes.newsPic}/>
            </a>
            <div className={classes.newsListItem}>
              <span className={classes.text}>#Video</span>
              <button className={classes.btnNewsListItem}>Follow</button>
            </div>
          </li>
        </div>
        <div className={classes.recommendations}>
          <span className={classes.recommendationsText}>View All Recommendations</span>
        </div>
      </div>
    </div>
  )
}

export default MainRight