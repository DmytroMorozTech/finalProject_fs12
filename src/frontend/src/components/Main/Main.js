import React from 'react'
import StyleMain from "./StyleMain";
import {MainRight} from "./MainRight";
import {MainLeft} from "./MainLeft";
import {Feed} from "./Feed/Feed";

function Main() {
    const classes = StyleMain();

    return (
        <div classes={classes.container}>
            <Section>
                <MainLeft/>
                <Feed/>
                <MainRight/>
            </Section>
        </div>
    )
}

export default Main;