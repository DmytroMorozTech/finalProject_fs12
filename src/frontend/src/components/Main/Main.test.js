import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import Main from './Main'

const setUp = (props) => shallow(<Main {...props}/>)

describe('should render Main component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
