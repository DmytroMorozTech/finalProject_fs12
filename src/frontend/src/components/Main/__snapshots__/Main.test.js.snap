// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Main component to match snapshot 1`] = `<Preloader />`;
